src/aggregate/aggregate_exec.c

```diff
@@ -1111,9 +1127,9 @@ static int execCommandCommon(RedisModuleCtx *ctx, RedisModuleString **argv, int
   return REDISMODULE_OK;
 
 error:
-  // Update global query errors statistics before freeing the request.
-  // Both SA and internal are considered shards.
-  QueryErrorsGlobalStats_UpdateError(QueryError_GetCode(&status), 1, SHARD_ERR_WARN);
+  // Update global query errors statistics
+  // If num shards == 1 we are in SA
+  QueryErrorsGlobalStats_UpdateError(QueryError_GetCode(&status), 1, GetNumShards_UnSafe() == 1);
```

Comment:
can't we use `AREQ` here instead of `GetNumShards_UnSafe()`?

Reply:
Using `GetNumShards_UnSafe()` enables checking without the `AREQ` which might be freed at this point